src/test/java/org/jabref/logic/bibtexkeypattern/BibtexKeyPatternUtilTest.java

```diff
@@ -57,6 +58,21 @@ public void testAndInAuthorName() throws ParseException {
     }
 
     @Test
+    public void testCrossrefAndInAuthorNames() throws ParseException {
+        BibDatabase database = new BibDatabase();
+        BibEntry entry1 = new BibEntry();
+        entry1.setField(FieldName.CROSSREF, "entry2");
+        database.insertEntry(entry1);
+        String bibtexString = "@ARTICLE{entry2, author={Simon Holland}}";
```

Comment:
Please construct the entry by hand (as you did for `entry1`) instead of using the parser. I know, the other tests also work like that but this is bad design.